src/prpy/tsr/generic.py

```diff
@@ -0,0 +1,275 @@
+import numpy
+from prpy.tsr.tsrlibrary import TSRFactory
+from prpy.tsr.tsr import TSR, TSRChain
+
+def get_manip_idx(robot, manip=None):
+    ''' 
+    Helper function for getting the manipulator index
+    to be used by the TSR
+    '''
```

Comment:
Add docstrings for arguments. Also use `"""` instead of `'''` for multi-line comments.